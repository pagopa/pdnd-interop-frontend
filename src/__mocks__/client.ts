import { Client } from '../../types'

export const clientActiveAgreementActiveEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementActiveEserviceDeprecated: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'DEPRECATED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
    activeDescriptor: {
      id: 'sdjfsdjf-djfsdj-sdjfdsj',
      state: 'PUBLISHED',
      version: '7',
    },
  },
  purposes: 'accesso pieno',
}

export const clientSuspendedAgreementActiveEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'SUSPENDED',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementSuspendedEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'SUSPENDED',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementActiveEserviceSuspended: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'SUSPENDED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}
