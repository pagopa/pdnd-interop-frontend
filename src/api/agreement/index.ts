export * from './agreement.mutations'
export * from './agreement.queries'
export * from './agreement.downloads'
export * from './agreement.services'
