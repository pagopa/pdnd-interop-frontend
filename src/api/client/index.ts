export * from './client.queries'
export * from './client.mutations'
export * from './client.services'
export * from './client.downloads'
