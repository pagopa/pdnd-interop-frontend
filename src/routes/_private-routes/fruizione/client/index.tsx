import React from 'react'
import { createFileRoute } from '@tanstack/react-router'

export const Route = createFileRoute('/_private-routes/fruizione/client/')({
  staticData: {
    authLevels: ['admin', 'support', 'security'],
    routeKey: 'SUBSCRIBE_CLIENT_LIST',
  },
  component: () => (
    <div>
      Hello /_authentication-guard/_tos-guard/_app-layout/_authorization-guard/fruizione/client/!
    </div>
  ),
})
