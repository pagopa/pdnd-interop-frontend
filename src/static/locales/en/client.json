{
  "noMultiDataLabel": "No clients available",
  "tableClientInPurpose": {
    "actions": {
      "removeFromPurpose": "Remove from purpose"
    },
    "noClientsAssociatedToPurposeLabel": "No clients associated with this purpose"
  },
  "create": {
    "title": "Create new client",
    "description": "Once the client is created, you can always add and remove operators and public keys to use for accessing producers' e-services",
    "infoSectionTitle": "General information",
    "nameField": {
      "label": "Client name (required)",
      "infoLabel": "Helps you distinguish it from others. Min 5 characters, max 60 characters"
    },
    "descriptionField": {
      "label": "Client description (required)",
      "infoLabel": "Min 10 characters, max 250 characters"
    },
    "clientOperatorsSection": {
      "title": "Client members",
      "description": "You can manage the client members at any time, even after creation"
    },
    "operatorsTable": {
      "noDataLabel": "No user added"
    },
    "addOperatorsDrawer": {
      "title": "Add members to client",
      "autocompleteInput": {
        "label": "Selected users"
      },
      "adminAlert": "If you want to upload public keys for this client, also add yourself to the client"
    },
    "addPublicKeyDrawer": {
      "title": "Enter new public key",
      "description": "Don't know how to create a set of cryptographic material? <1>Read the guide</1>.",
      "nameInput": {
        "label": "Key name (required)",
        "infoLabel": "It will help you distinguish it from others. Min 5 characters, max 60 characters"
      },
      "publicKeyInput": {
        "label": "Public key (required)"
      },
      "insertAction": "Insert"
    },
    "actions": {
      "createLabel": "Create",
      "backToClientsLabel": "Back to clients"
    }
  },
  "edit": {
    "alertApiV2": {
      "description": "PDND will soon update its API model to version 2. In the future, it will no longer be possible to interact with the previous version.",
      "linkLabel": "Learn more"
    },
    "adminSection": {
      "title": "Admin API",
      "description": "The Client Admin API is a user belonging to the party who is authorized to perform write operations through the machine-to-machine channel.",
      "adminLabel": "Manager",
      "actions": {
        "selectAdminLabel": "Select admin",
        "substituteAdminLabel": "Replace",
        "removeAdminLabel": "Remove"
      }
    },
    "setClientAdminDrawer": {
      "title": {
        "substitute": "Replace Client Admin API",
        "select": "Select Client Admin API"
      },
      "subtitle": "You can enable a client member for writing via API in the machine-to-machine flow.",
      "adminField": {
        "label": "Client Admin API",
        "infoLabel": "You can choose only one member per client.",
        "validation": {
          "sameValue": "The Client Admin API must be different from the current one"
        }
      }
    },
    "tabs": {
      "ariaLabel": "Three different tabs for voucher acquisition instructions, client members, and public keys",
      "voucher": "How do I get a voucher?",
      "clientOperators": "Client members",
      "publicKeys": "Public keys"
    },
    "actions": {
      "backToClientsLabel": "Back to clients list"
    },
    "filters": {
      "operatorField": {
        "label": "Search by user"
      }
    }
  },
  "list": {
    "filters": {
      "nameField": {
        "label": "Search by name"
      }
    }
  }
}
