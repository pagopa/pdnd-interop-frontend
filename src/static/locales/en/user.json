{
  "backToMemberListBtn": "Back to members list",
  "clientOperatorsTab": {
    "title": "Client members",
    "description": "Members belonging to this client will appear in the list. If an API Admin is appointed, each member will be able to perform write operations."
  },
  "actionReservedToAdminTooltip": "This actions is reserved to user with administrator permissions",
  "actions": {
    "removeFromClient": {
      "label": "Remove from client",
      "tooltip": "The operation is permitted only to users with administrator permissions"
    },
    "delete": {
      "label": "Delete",
      "tooltip": "The operation is permitted only to users with administrator permissions"
    }
  },
  "edit": {
    "generalInformations": "General informations",
    "productRoleField": {
      "label": "Permissions on PDND Interoperability"
    },
    "operatorKeysLink": {
      "label": "Go to the view of the keys associated with the user"
    }
  }
}
