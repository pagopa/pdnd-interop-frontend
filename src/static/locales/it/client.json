{
  "loadingSingleLabel": "Stiamo caricando il client",
  "loadingMultiLabel": "Stiamo caricando la lista dei client",
  "noMultiDataLabel": "Non ci sono client disponibili",
  "tableClientInPurpose": {
    "actions": {
      "removeFromPurpose": "Rimuovi dalla finalità"
    },
    "noClientsAssociatedToPurposeLabel": "Non ci sono client associati a questa finalità"
  },
  "create": {
    "title": "Crea nuovo client",
    "description": "Una volta creato il client, potrai sempre aggiungere e rimuovere operatori e chiavi pubbliche da utilizzare per accedere agli e-service degli erogatori",
    "infoSectionTitle": "Informazioni generali",
    "nameField": {
      "label": "Nome del client (richiesto)",
      "infoLabel": "Ti aiuta a distinguerlo dagli altri. Min 5 caratteri, max 60 caratteri"
    },
    "descriptionField": {
      "label": "Descrizione del client (richiesto)",
      "infoLabel": "Min 10 caratteri, max 250 caratteri"
    },
    "clientOperatorsSection": {
      "title": "Membri del client",
      "description": "Potrai gestire i membri del client in ogni momento, anche dopo la creazione"
    },
    "operatorsTable": {
      "noDataLabel": "Nessun utente aggiunto"
    },
    "addOperatorsDrawer": {
      "title": "Aggiungi membri al client",
      "autocompleteInput": {
        "label": "Utenti selezionati"
      },
      "adminAlert": "Se vuoi caricare chiavi pubbliche per questo client, aggiungi anche te stesso al client"
    },
    "addPublicKeyDrawer": {
      "title": "Inserisci nuova chiave pubblica",
      "description": "Non sai come fare per creare un set di materiale crittografico? <1>Consulta la guida</1>.",
      "nameInput": {
        "label": "Nome della chiave (richiesto)",
        "infoLabel": "Ti aiuterà a distinguerla dalle altre. Min. 5 caratteri, max. 60 caratteri"
      },
      "publicKeyInput": {
        "label": "Chiave pubblica (richiesto)"
      },
      "insertAction": "Inserisci"
    },
    "actions": {
      "createLabel": "Crea",
      "backToClientsLabel": "Torna ai client"
    }
  },
  "edit": {
    "tabs": {
      "ariaLabel": "Tre tab diverse per le istruzioni dell'ottenimento del voucher, i membri del client e le chiavi pubbliche",
      "voucher": "Come ottengo un voucher?",
      "clientOperators": "Membri del client",
      "publicKeys": "Chiavi pubbliche"
    },
    "actions": {
      "deleteLabel": "Elimina client",
      "backToClientsLabel": "Torna alla lista dei client"
    },
    "filters": {
      "operatorField": {
        "label": "Cerca per utente"
      }
    }
  },
  "list": {
    "filters": {
      "nameField": {
        "label": "Cerca per nome"
      }
    },
    "consumer": {
      "title": "I tuoi client e-service",
      "description": "In quest'area puoi i trovare e gestire tutti i client e-service che hai creato"
    }
  },
  "interopM2M": {
    "title": "I tuoi client API Interop",
    "description": "Gestisci i client per staccare voucher validi per interrogare le API di Interoperabilità"
  }
}
