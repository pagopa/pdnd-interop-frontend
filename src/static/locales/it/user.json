{
  "loadingSingleLabel": "Stiamo caricando l'operatore",
  "loadingMultiLabel": "Stiamo caricando la lista degli operatori",
  "noMultiDataLabel": "Non ci sono operatori disponibili",
  "backToMemberListBtn": "Torna alla lista membri",
  "actionReservedToAdminTooltip": "Quest’azione è riservata agli utenti con permessi da amministratore",
  "actions": {
    "removeFromClient": {
      "label": "Rimuovi dal client",
      "tooltip": "L’operazione è consentita solo agli utenti con permessi da amministratore"
    },
    "delete": {
      "label": "Elimina",
      "tooltip": "L’operazione è consentita solo agli utenti con permessi da amministratore"
    }
  },
  "edit": {
    "generalInformations": "Informazioni generali",
    "taxCodeField": {
      "label": "Codice fiscale"
    },
    "roleField": {
      "label": "Ruolo"
    },
    "productRoleField": {
      "label": "Permessi su PDND Interoperabilità"
    },
    "statusField": {
      "label": "Stato dell'utenza"
    },
    "associatedKeysField": {
      "label": "Chiavi associate",
      "noDataLabel": "Nessuna chiave associata"
    },
    "operatorKeysLink": {
      "label": "Vai alla vista delle chiavi associate all’utente"
    }
  },
  "list": {
    "provider": {
      "title": "Operatori API dell'ente",
      "description": "In quest’area puoi trovare e gestire tutti gli operatori API che sono stati abilitati alla gestione degli e-service dell'ente",
      "tempAlertMessage": "Se l'operatore non è in elenco, in questa fase di test contattaci per aggiungerlo"
    }
  }
}
