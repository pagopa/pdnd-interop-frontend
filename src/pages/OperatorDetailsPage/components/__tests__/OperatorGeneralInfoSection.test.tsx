import React from 'react'
import { render } from '@testing-library/react'
import {
  OperatorGeneralInfoSection,
  OperatorGeneralInfoSectionSkeleton,
} from '../OperatorGeneralInfoSection'
import type { RelationshipInfo } from '@/api/api.generatedTypes'
import { ClientQueries } from '@/api/client'
import { vi } from 'vitest'
import { createMockSelfCareUser } from '__mocks__/data/user.mocks'

const mockUseGetSingleOperator = (result: RelationshipInfo | undefined) => {
  vi.spyOn(ClientQueries, 'useGetSingleOperator').mockReturnValue({
    data: result,
  } as unknown as ReturnType<typeof ClientQueries.useGetSingleOperator>)
}

describe('OperatorGeneralInfoSection', () => {
  it('should match the snapshot', () => {
    mockUseGetSingleOperator(createMockSelfCareUser())
    const { baseElement } = render(<OperatorGeneralInfoSection operatorId="operatorId" />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should not render if the operator data is undefined', () => {
    mockUseGetSingleOperator(undefined)
    const { container } = render(<OperatorGeneralInfoSection operatorId="operatorId" />)
    expect(container).toBeEmptyDOMElement()
  })
})

describe('OperatorGeneralInfoSectionSkeleton', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(<OperatorGeneralInfoSectionSkeleton />)
    expect(baseElement).toMatchSnapshot()
  })
})
