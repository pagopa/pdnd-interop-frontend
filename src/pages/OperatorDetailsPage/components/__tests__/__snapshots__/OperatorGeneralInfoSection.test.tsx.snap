// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OperatorGeneralInfoSection > should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          generalInformations
        </h2>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="css-1p5q5e5-MuiStack-root"
        >
          <div
            class="css-88r4rx-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                taxCodeField.label
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                MRORSSR78M21B354T
              </p>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                productRoleField.label
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                userProductRole.admin
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`OperatorGeneralInfoSectionSkeleton > should match the snapshot 1`] = `
<body>
  <div>
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i8vzz-MuiSkeleton-root"
      style="height: 235px;"
    />
  </div>
</body>
`;
